### PR TITLE
Tag filter for admin tags index page

### LINKED ISSUE #1695: Put in filter on admin tags index page
As an admin user who is looking at tags on the tags index page I would like some filters/pagination so I can better locate the tag I am interested in.

## Acceptance Criteria

- [x] Filter by type (Facility, Partnership or Category)
- [x] Basic name search
- [x] Pagination (page number and per-page limit)

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Closes #1695 

Filter by
- Type
- Name (very crudely)
- Pagination

### CODE REVIEW

File: test/controllers/concerns/tag_filter_test.rb [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+      name: 'alpha',
+      per_page: '20',
+      page_num: '2'
+    }
+    filter = TagFilter.new(params)
+
+    output = filter.next_page_link(FakeView.new, Tag.all)
+    assert output.is_a?(String)
+
+    expected_html = '<a class="btn btn-link" href="/tags?name=alpha&page_num=3&per_page=20&type=facility">Next ...</a>'
```
Comment (aaaaargZombies) [inline test/controllers/concerns/tag_filter_test.rb:216]: Can we make this a button and add a "previous" if we've already clicked "next?

**Review by aaaaargZombies — APPROVED** [review]
works nicely - I left a comment for a possible improvement that I'll leave up to you.

File: test/controllers/concerns/tag_filter_test.rb (hunk 1 above)
Comment (ivan-kocienski-gfsc) [reply test/controllers/concerns/tag_filter_test.rb:216]: Maybe? But there is a back button on their browser (hope that doesn't come off as too sarcastic).

When ever I see things like this one sites that do the whole pagination circus with `<< < 0 2 3 ... 45 > >>` type things i think about how i only really use the next page link.

Comment (aaaaargZombies) [reply test/controllers/concerns/tag_filter_test.rb:216]: > Maybe? But there is a back button on their browser

happy with this

I do think next would benefit from looking like a button though but the PR is still good overall

**Pull request merged**